### PR TITLE
Add Skill Models

### LINKED ISSUE #1539: My Skills - Create skill families in the backend
# Description

Skill Families are groups of skills, and fall into one of several Skill Categories. In the graphql schema, they can be modeled like this:
```
enum SkillCategory {
  TECHNICAL
  BEHAVIOURAL
}

type SkillFamily {
  id: ID!
  name: LocalizedString!
  description: LocalizedString!
  skills: [Skill]
  category: SkillCategory!
}
```
The SkillFamily relationship should also be added to the Skill definition in the graphql schema:
```
type Skill {
  ...
  families: [SkillFamily]
}
```
Skills to SkillFamilies is a many-to-many relationship, as a few skills can belong to multiple overlapping families. This can be accomplished in the database with a pivot table.

# Definition of done
- [ ] Migration which adds SkillFamilies table to DB
- [ ] Migration which adds skill_skill_family pivot table
- [ ] Eloquent model for SkillFamily
- [ ] Add family relationship to Skill eloquent model
- [ ] A SkillFamilyFactory, linked to Eloquent model.
- [ ] DevSeeder should ensure all skills have an associated SkillFamily. Each family should have 5-10 skills.
- [ ] New types added to graphql schema. Skill type updated in schema.

### LINKED ISSUE #1540: My Skills - Create Skill Models in the backend
# Description
Add the Skill model to the API server.
In the graphql schema, the skill type should look like this:

```
type Skill {
  id: ID!
  key: KeyString!
  name: LocalizedString!
  description: LocalizedString
  keywords: [String]
}
```

Keywords should be implemented similar to roles in the User model: cast to an array in the Eloquent model, and implemented as a jsonb field in the database.

The skill model itself is pretty simple. Most of its complexity will come from its relationships with many other models (Experiences, Applicant Profiles, Skill Categories...) but those are later tasks, dependent on those other models existing first.

# Definition of done
- [ ] Migration which adds Skills table to DB
- [ ] Eloquent model for Skill
- [ ] SkillFactory, linked to Eloquent model
- [ ] DevSeeder uses SkillFactory to seed some number of random skills
- [ ] Skill type added to graphql schema

### PR DESCRIPTION
**Pull request by petertgiles**
This branch is the start of a back-end implementation of the skills hierarchy.  It implements two new models:
- SkillFamily
- Skill

And one enum:
- SkillCategory

The many-to-many relationship between the two is implemented and the query to list all instances of each is also implemented.  The relationship to the user model will be implemented in #1542 and the ability to mutate the relationship with a user will be implemented in #1541.

To test, run the migration and seeders: `php artisan migrate --seed`

Here are some queries to test the branch:
```
{
  skillFamilies {
    id
    key
    name { en }    
    description { en }
    skills {
      name { en }
    }
    category
  }
}

{
  skills {
    id
    key
    name { en }
    description { en }
    keywords
    families { 
      name { en }
    }
  }
}
```

Closes #1539 
Closes #1540

### CODE REVIEW

**Review by brindasasi — APPROVED** [review]
Looks good to me!

**Review by yonikid15 — COMMENTED** [review]
Nice work Peter, and this looks correct. However, I don't see any descriptions/acceptance criteria for the issues you completed, so I'm not 100% sure.

File: api/graphql/schema.graphql [hunk 1]
```diff
@@ -189,6 +189,29 @@ type PoolCandidateSearchRequest {
...
+    TECHNICAL,
+    BEHAVIOURAL
+}
+
+type SkillFamily {
+    id: ID!
+    key: KeyString!
+    name: LocalizedString!
+    description: LocalizedString!
+    skills: [Skill] @hasMany
```
Comment (yonikid15) [inline api/graphql/schema.graphql:202]: Shouldn't this and line 212 have the directive @belongsToMany instead?

Comment (petertgiles) [reply api/graphql/schema.graphql]: Can't get anything past you!  Great catch. 😀

**Review by yonikid15 — APPROVED** [review]
Lgtm, nice work as usual 👍

**Pull request merged**